packages/@aws-cdk/aws-codebuild/lib/project.ts

```diff
@@ -727,7 +727,7 @@ export class Project extends ProjectBase {
         // save the SSM env variables
         if (envVariable.type === BuildEnvironmentVariableType.PARAMETER_STORE) {
           const envVariableValue = envVariable.value.toString();
-          ssmVariables.push(Stack.of(principal).formatArn({
+          ssmVariables.push(Stack.of(principal as unknown as IConstruct).formatArn({
```

Comment:
@skinny85 - this is probably incorrect but was the quickest way to resolve this error.

After merging this PR, could you follow up with the correct fix?

Reply:
Sure, will do!